core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -295,12 +295,12 @@ private[spark] class Executor(
 
         // Deserialization happens in two parts: first, we deserialize a Task object, which
         // includes the Partition. Second, Task.run() deserializes the RDD and function to be run.
-        task.metrics.setExecutorDeserializeTime(
+        task.metrics.incExecutorDeserializeTime(
           (taskStart - deserializeStartTime) + task.executorDeserializeTime)
         // We need to subtract Task.run()'s deserialization time to avoid double-counting
-        task.metrics.setExecutorRunTime((taskFinish - taskStart) - task.executorDeserializeTime)
-        task.metrics.setJvmGCTime(computeTotalGcTime() - startGCTime)
-        task.metrics.setResultSerializationTime(afterSerialization - beforeSerialization)
+        task.metrics.incExecutorRunTime((taskFinish - taskStart) - task.executorDeserializeTime)
```

Comment:
Yes. But I'd still remove the setValue, and just call reset().add(..), once we decouple copyAndReset into copy and reset.